lnbits/app.py

```diff
@@ -68,6 +69,52 @@
 )
 
 
+@asynccontextmanager
+async def lifespan(app: FastAPI):
+    # wait till migration is done
+    await migrate_databases()
+
+    # setup admin settings
+    await check_admin_settings()
+    await check_webpush_settings()
+
+    log_server_info()
+
+    # initialize WALLET
+    try:
+        set_wallet_class()
+    except Exception as e:
+        logger.error(f"Error initializing {settings.lnbits_backend_wallet_class}: {e}")
+        set_void_wallet_class()
+
+    # initialize funding source
+    await check_funding_source()
+
+    # register core routes
+    init_core_routers(app)
+
+    # check extensions after restart
+    if not settings.lnbits_extensions_deactivate_all:
+        await check_installed_extensions(app)
+        register_all_ext_routes(app)
+
+    if settings.lnbits_admin_ui:
+        initialize_server_logger()
+
+    # initialize tasks
+    register_async_tasks()
+
+    yield
```

Comment:
Suggestion: clearly separate the `start-up` and `shut-down` logic into two different functions.

```py
@asynccontextmanager
async def lifespan(app: FastAPI):
     await lifespan_startup(app)
     yield
     await lifespan_shutdown(app)
```